gatling-datadog/src/main/scala/io/gatling/datadog/DatadogRequests.scala

```diff
@@ -1,249 +1,262 @@
-      }
-    }
+    // The specification for Datadog histogram data is given here
+    // https://docs.datadoghq.com/metrics/types/?tab=histogram#metric-types
 
-    requestLatencyRequests.foreach { requestLatencyRequest =>
-      requestLatencyRequest.onComplete {
-        case Success(StatusResponse(status)) =>
-          requestLatency
-            .keys()
-            .asScala
-            .toSeq
-            .foreach(id => DatadogData.remove(id, requestLatency))
-        case Failure(e) =>
-          logger.error(s"Request latency request failed with error $e")
-      }
-    }
+    val values = dataPoints.map(_.value.doubleValue)
 
-    errorCounterRequests.foreach { errorCounterRequest =>
-      errorCounterRequest.onComplete {
-        case Success(StatusResponse(status)) =>
-          errorCounter
-            .keys()
-            .asScala
-            .toSeq
-            .foreach(id => DatadogData.remove(id, errorCounter))
-        case Failure(e) =>
-          logger.error(s"Error count request failed with error $e")
-      }
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.avg",
+      metricType = MetricType.Gauge,
+      value = values.sum / dataPoints.length,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.count",
+      metricType = MetricType.Rate,
+      value = dataPoints.length,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.median",
+      metricType = MetricType.Gauge,
+      value = median(values),
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.max",
+      metricType = MetricType.Gauge,
+      value = values.max,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.min",
+      metricType = MetricType.Gauge,
+      value = values.max,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.95percentile",
+      metricType = MetricType.Gauge,
+      value = p95(values),
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.99percentile",
+      metricType = MetricType.Gauge,
+      value = p99(values),
+      tags = tags
+    )
+  }
+
+  private def median(values: Seq[Double]): Double = {
+    val sortedList = values.sorted
+    val mid = values.size / 2
+    if (values.size % 2 == 0) {
+      (sortedList(mid - 1) + sortedList(mid)) / 2
+    } else {
+      sortedList(mid)
     }
   }
 
-  def send(
+  private def p95(values: Seq[Double]): Double = {
+    // calculate the 95th percentile
+    val sortedList = values.sorted
+    val index = (sortedList.size * 0.95).toInt - 1
+    sortedList(index)
+  }
+  private def p99(values: Seq[Double]): Double = {
+    // calculate the 95th percentile
+    val sortedList = values.sorted
+    val index = (sortedList.size * 0.99).toInt - 1
+    sortedList(index)
+  }
```

Comment:
(not important) comment has 95th percentile for both, this could be extracted into a generic function where you pass in the desired percentile